pkg/webhook/pod/tikv_creater.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pod
+
+import (
+	"fmt"
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/pdapi"
+	"github.com/pingcap/tidb-operator/pkg/webhook/util"
+	admission "k8s.io/api/admission/v1"
+	core "k8s.io/api/core/v1"
+	"k8s.io/klog"
+	"strings"
+)
+
+func (pc *PodAdmissionControl) admitCreateTiKVPod(pod *core.Pod, tc *v1alpha1.TidbCluster, pdClient pdapi.PDClient) *admission.AdmissionResponse {
+
+	name := pod.Name
+	namespace := pod.Namespace
+
+	stores, err := pdClient.GetStores()
+	if err != nil {
+		klog.Infof("Failed to get stores during pod [%s/%s] creation, error: %v", namespace, name, err)
+		return util.ARFail(err)
+	}
+	evictLeaderSchedulers, err := pdClient.GetEvictLeaderSchedulers()
+	if err != nil {
+		klog.Infof("failed to create pod[%s/%s],%v", namespace, name, err)
+		return util.ARFail(err)
```

Comment:
When TiKV is not bootstrapped, PD will respond an error when the client accesses `/pd/api/v1/schedulers` which is the endpoint we are visiting in `GetEvictLeaderSchedulers` method.
So will the cluster never get bootstrapped if we block tikv pod creation?

Reply:
> When TiKV is not bootstrapped, PD will respond an error when the client accesses `/pd/api/v1/schedulers` which is the endpoint we are visiting in `GetEvictLeaderSchedulers` method.
> So will the cluster never get bootstrapped if we block tikv pod creation?

In `AdmitCreatePods` func in `pkg/webhook/pod/pods.go` would let the request pass directly when `ownerStatefulSet.Status.CurrentReplicas < 1 && ownerStatefulSet.Status.UpdatedReplicas < 1 && ordinal == 0` .  In this case, if we create tikv with 3 replicas , the request for creating tikv-0, tikv-1,tikv-2 would be sent at the same time. The tikv-0 would be admited to created while tikv-1 and tikv-2 would be refused to created as they couldn't visit the tikv store Info. When tikv-0 started successfully, tikv-1 and tikv-2 would be created successfully.

Reply:
This is unreasonable, why tikv needs a seed pod? The way tikv works shouldn't rely on this. What do you think? @weekface

Reply:
The main problem is that what should we do if we couldn't visit tikv store info (like http status error ) when we receive creating tikv pod request. If we refuses all the requests, then we couldn't create tikv during tikv bootstrapped. If we admit all the requests, then we couldn't end evict leader for former pod.

Reply:
Now, `GetEvictLeaderSchedulers` will return an error when the TiKV cluster was not bootstrapped.

We should change the method `GetEvictLeaderSchedulers` to return an empty array(means there are no schedules) when TiKV was bootstrapping.

We needn't a seed pod.

Reply:
> Now, `GetEvictLeaderSchedulers` will return an error when the TiKV cluster was not bootstrapped.
> 
> We should change the method `GetEvictLeaderSchedulers` to return an empty array(means there are no schedules) when TiKV was bootstrapping.
> 
> We needn't a seed pod.

I think this method is better. I would accept it and remove seed pod method.